# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../dropbear/legacy_sus.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

usleep() was deprecated by SUSv3 and newer LIBCs are starting to enforce that
deprecation.

--- ./svr-auth.c.orig	2010-09-20 16:12:52.000000000 -0400
+++ ./svr-auth.c	2010-09-20 16:15:08.000000000 -0400
@@ -337,7 +337,8 @@
 	encrypt_packet();
 
 	if (incrfail) {
-		usleep(300000); /* XXX improve this */
+		const struct timespec ts = { 0, 300000000L };
+		nanosleep(&ts, NULL); /* XXX improve this */
 		ses.authstate.failcount++;
 	}
 
